### PR TITLE
chore(system-e2e): Support for mocking x-road Properties/assets api calls

### PR DESCRIPTION
**Pull request by pshomov**
# ...

Tuning configuration (mostly) so we can run service-portal with wire-mocks for X-Road. This is in order to support system e2e tests.

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/service-portal/graphql/src/lib/client.ts [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
+  HttpLink,
+  InMemoryCache,
 } from '@apollo/client'
 import { onError } from '@apollo/client/link/error'
 import { RetryLink } from '@apollo/client/link/retry'
 import { authLink } from '@island.is/auth/react'
+import { getStaticEnv } from '@island.is/shared/utils'
 
 const uri =
-  process.env.NODE_ENV === 'development'
```
Comment (pshomov) [inline libs/service-portal/graphql/src/lib/client.ts:13]: Hey @brynjarorng, do you think this is an issue, security-wise?

**Review by Thorrik58 — APPROVED** [review]
Looks good from the application teams files

File: apps/api/infra/api.ts [hunk 2]
```diff
@@ -161,10 +161,15 @@ export const serviceSetup = (services: {
       },
       NO_UPDATE_NOTIFIER: 'true',
       FISKISTOFA_ZENTER_CLIENT_ID: '1114',
+      SOFFIA_SOAP_URL: {
+        dev: ref((h) => h.svc('https://soffiaprufa.skra.is')),
+        staging: ref((h) => h.svc('https://soffiaprufa.skra.is')),
+        prod: ref((h) => h.svc('https://soffia.skra.is')),
```
Comment (brynjarorng) [inline apps/api/infra/api.ts:167]: Should be `soffia2.skra.is`

**Review by brynjarorng — APPROVED** [review]
Please look into changing the soffia url to the new one. Otherwise LGTM

File: apps/system-e2e/src/support/wire-mocks.ts [hunk 3]
```diff
@@ -9,17 +9,64 @@ import {
...
+    const result = envValues[env]
+    if (typeof result === 'string') {
+      throw new Error('Should have been a reference, not a string value')
+    } else {
+      return result
+    }
+  } else if (typeof envVar === 'function') {
+    return envVar
+  } else {
+    throw new Error('Env variable should have been a reference')
```
Comment (kristofer-lp8f) [inline apps/system-e2e/src/support/wire-mocks.ts:64]: This could be rewritten using guard patterns for less indentation. Many `if...else return/throw`

File: infra/src/dsl/value-files-generators/local-setup.ts [hunk 4]
```diff
@@ -78,6 +78,14 @@ export const getLocalrunValueFile = async (
                     proxy: {
                       to: target.replace('localhost', 'host.docker.internal'),
```
Comment (kristofer-lp8f) [inline infra/src/dsl/value-files-generators/local-setup.ts:79]: Will this work with `podman` as well?

File: infra/src/dsl/xroad.ts [hunk 5]
```diff
@@ -26,20 +26,22 @@ export type XRoadEnvs<
...
+}
 export const Base = new XroadConf({
   env: {
-    XROAD_BASE_PATH: {
-      dev: ref((h) => h.svc('http://securityserver.dev01.devland.is')),
-      staging: 'http://securityserver.staging01.devland.is',
-      prod: 'http://securityserver.island.is',
-      local: ref((h) => h.svc('http://localhost:8081')),
-    },
+    XROAD_BASE_PATH: XROAD_BASE_PATH,
```
Comment (kristofer-lp8f) [inline infra/src/dsl/xroad.ts:37]: Why did you move the config?

File: libs/shared/utils/src/lib/createXRoadAPIPath.ts [hunk 6]
```diff
@@ -1,5 +1,3 @@
-import { logger } from '@island.is/logging'
-
```
Comment (kristofer-lp8f) [inline libs/shared/utils/src/lib/createXRoadAPIPath.ts:2]: Could the logging module be monkey-patched out instead? We will likely run into more logger issues in the future, similar to this.

**Review by kristofer-lp8f — APPROVED** [review]
LGTM

File: infra/src/dsl/value-files-generators/local-setup.ts (hunk 4 above)
Comment (pshomov) [reply infra/src/dsl/value-files-generators/local-setup.ts:79]: No idea about that.

File: libs/shared/utils/src/lib/createXRoadAPIPath.ts (hunk 6 above)
Comment (pshomov) [reply libs/shared/utils/src/lib/createXRoadAPIPath.ts:2]: I don't know about that. I kind of avoid monkey-patching ;)

Comment (eirikurn) [reply libs/shared/utils/src/lib/createXRoadAPIPath.ts:2]: I want to get rid of this function, but in another branch we changed it to use console.error instead to get rid of this dependency.

**Review by eirikurn — APPROVED** [review]

**Review by thordurhhh — CHANGES_REQUESTED** [review]
Service-portal looks good. Just that suggestion from Brynjar that needs updating https://github.com/island-is/island.is/pull/9556/files#r1067951189

File: infra/src/dsl/xroad.ts (hunk 5 above)
Comment (pshomov) [reply infra/src/dsl/xroad.ts]: Great point, I was moving along so that was the easiest thing to do. Made an improvement.

File: apps/system-e2e/src/support/wire-mocks.ts (hunk 3 above)
Comment (pshomov) [reply apps/system-e2e/src/support/wire-mocks.ts]: Refactored to a switch statement. Somewhat better I feel. Feel free to make a suggestion if you will ;)

File: apps/api/infra/api.ts (hunk 2 above)
Comment (pshomov) [reply apps/api/infra/api.ts]: fixed

File: libs/service-portal/graphql/src/lib/client.ts (hunk 1 above)
Comment (pshomov) [reply libs/service-portal/graphql/src/lib/client.ts:13]: Ping @brynjarorng ^

**Review by thordurhhh — APPROVED** [review]
Nice 🙌 🤩
